File: src/main/java/com/leesh/inflpick/product/domain/ProductId.java
package com.leesh.inflpick.product.domain;

import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public record ProductId(@NotNull String uuid) {

    public ProductId {
        Objects.requireNonNull(uuid, "uuid must not be null");
        if (uuid.isBlank()) {
            throw new IllegalArgumentException("uuid must not be blank");
        }
    }
}

File: src/main/kotlin/com/leesh/inflpick/product/domain/ProductId.kt
package com.leesh.inflpick.product.domain

import com.leesh.inflpick.common.domain.DomainId

class ProductId(private val id: String): DomainId {
    override fun isSaved(): Boolean {
        return id.isNotEmpty()
    }

    override fun toString(): String {
        return id
    }

    companion object {
        fun fromString(id: String): DomainId {
            return ProductId(id)
        }
    }
}